### PR TITLE
nest 2.12.0

### PR DESCRIPTION
**Pull request by tammoippen**
Update to latest version nest 2.12.0

- Remove (unnecessary) patching.
- Add Python3 option.
- Link to correct Python2 libraries (and use macOS or brewed python).
- Remove autotools build chain, as we moved to cmake.
- Some audit cleanups.

### Have you:

- [X] Followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-science/blob/master/.github/CONTRIBUTING.md) document?
- [X] Checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-science/pulls) for the same update/change?
- [X] Run `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting) and corrected all errors?
- [X] Built your formula locally prior to submission with `brew install <formula>`?

### Updates to existing formula: have you

- [X] Removed the `revision` line, if any, when bumping the version number?
- [X] Added/bumped the `revision` number if the changes affect the precompiled bottles?
- [X] Not altered the `bottle` section?
- [X] Checked that the tests still pass?

### CODE REVIEW

File: nest.rb [hunk 1]
```diff
@@ -125,46 +95,25 @@ def install
...
+        [/numproc     /executable /scriptfile]
         {
          () [
-          (mpirun -np ) numproc cvs ( ) statusdict/prefix :: (/bin/nest )  slifile
+          (mpirun -np ) numproc cvs ( ) executable ( ) scriptfile
          ] {join} Fold
         } Function def
-      }
+      EOS
       File.open(ENV["HOME"]+"/.nestrc", "w") { |file| file.write(nestrc) }
```
Comment (jonchang) [inline nest.rb]: I'm surprised that writing to the user's home directory works inside the sandbox. Did you test this locally?

Comment (tammoippen) [inline nest.rb]: This does not write to the user `HOME`. It is actually the sandbox's `HOME`. NEST uses the environment var `HOME` as the default place to look for `.nestrc`, and we need a slightly modified`.nestrc`, when compiled with `mpi`.

File: nest.rb [hunk 2]
```diff
@@ -13,98 +12,69 @@ class Nest < Formula
...
   depends_on :mpi => [:optional, :cc, :cxx]
-  depends_on :python => :optional if MacOS.version <= :snow_leopard
+
+  # Any Python >= 2.7 < 3.x is okay (either from macOS or brewed)
+  depends_on :python => :optional
   depends_on "numpy" => :python if build.with? "python"
   depends_on "scipy" => :python if build.with? "python"
   depends_on "matplotlib" => :python if build.with? "python"
   depends_on "cython" => :python if build.with? "python"
   depends_on "nose" => :python if build.with? "python"
```
Comment (jonchang) [inline nest.rb]: These should be something like

```ruby
if build.with? "python"
  depends_on "foo"
end
```

File: nest.rb [hunk 3]
```diff
@@ -13,98 +12,69 @@ class Nest < Formula
...
 
   depends_on "gsl" => :recommended
   depends_on :mpi => [:optional, :cc, :cxx]
-  depends_on :python => :optional if MacOS.version <= :snow_leopard
+
+  # Any Python >= 2.7 < 3.x is okay (either from macOS or brewed)
+  depends_on :python => :optional
   depends_on "numpy" => :python if build.with? "python"
   depends_on "scipy" => :python if build.with? "python"
   depends_on "matplotlib" => :python if build.with? "python"
```
Comment (jonchang) [inline nest.rb]: `numpy`, `scipy`, and `matplotlib` should use the formula versions, not the python dependency. `brew info numpy` for details.

File: nest.rb [hunk 4]
```diff
@@ -13,98 +12,69 @@ class Nest < Formula
...
   depends_on "cython" => :python if build.with? "python"
   depends_on "nose" => :python if build.with? "python"
+
+  depends_on :python3 => :optional
+  depends_on "numpy" => :python3 if build.with? "python3"
+  depends_on "scipy" => :python3 if build.with? "python3"
+  depends_on "matplotlib" => :python3 if build.with? "python3"
+  depends_on "cython" => :python3 if build.with? "python3"
+  depends_on "nose" => :python3 if build.with? "python3"
```
Comment (jonchang) [inline nest.rb]: See above

**Comment by jonchang** [comment]
Thanks @tammoippen! This has been merged.

**Pull request closed without merging**